packages/framework/esm-api/src/shared-api-objects/location.ts

```diff
@@ -11,8 +11,16 @@ export function toLocationObject(openmrsRestForm: any): Location {
   };
 }
 
-export function getLocations(tagUuidOrName: string | null = null): Observable<Array<Location>> {
-  const url = `${restBaseUrl}/location` + (tagUuidOrName ? '?tag=' + tagUuidOrName : '');
+export function getLocations(
+  tagUuidOrName: string | null = null,
+  query: string | null = null,
+): Observable<Array<Location>> {
+  const url =
+    `${restBaseUrl}/location` +
+    (tagUuidOrName || query ? '?' : '') +
+    (tagUuidOrName ? 'tag=' + tagUuidOrName : '') +
+    (tagUuidOrName && query ? '&' : '') +
+    (query ? 'q=' + query : '');
   return openmrsObservableFetch<any>(url)
```

Comment:
```suggestion
  const params = new URLSearchParams();
  if (tagUuidOrName) {
    params.set('tag', tagUuidOrName);
  }
  if (query) {
    params.set('q', query);
  }
  const queryString = params.toString();
  const url = `${restBaseUrl}/location${queryString ? '?' + queryString : ''}`;

  return openmrsObservableFetch<{ results: Array<Location> }>(url)
```

Would something like this work as well? The current URL construction could produce an invalid URL when `tagUuidOrName` is null but query is provided. In this case, the URL would include an empty string where the tag= parameter should be, potentially resulting in malformed URLs like:

`/location?&q=searchterm`

Also, this annotates the return type of the locations response correctly.

Reply:
Perfect!  Thanks so much @denniskigen ... I *think* I handled the null case, but this is *much* cleaner... I will commit your change, test, and then merge if all is well...